pkg/skaffold/cluster/minikube.go

```diff
@@ -0,0 +1,204 @@
+	return clientImpl{}
+}
+
+func (clientImpl) IsMinikube(kubeContext string) (bool, error) {
+	// short circuit if context is 'minikube'
+	if kubeContext == constants.DefaultMinikubeContext {
+		return true, nil
+	}
+	_, err := mkBinaryFunc()
+	if err != nil {
+		return false, nil // minikube binary not found
+	}
+
+	if ok, err := matchNodeLabel(kubeContext); err != nil {
+		return false, nil
+	} else if ok {
+		logrus.Debugf("Detected minikube cluster for context %s due to matched labels", kubeContext)
+		return true, nil
+	}
+
+	if ok, err := matchProfileAndServerURL(kubeContext); err != nil {
+		return false, nil
+	} else if ok {
+		logrus.Debugf("Detected minikube cluster for context %s due to matched profile name or server url", kubeContext)
+		return true, nil
+	}
+	return false, nil
+}
+
+func (clientImpl) MinikubeExec(arg ...string) (*exec.Cmd, error) {
+	return minikubeExec(arg...)
+}
+
+func minikubeExec(arg ...string) (*exec.Cmd, error) {
+	b, err := mkBinaryFunc()
+	if err != nil {
+		return nil, fmt.Errorf("getting minikube executable: %w", err)
+	}
+	return exec.Command(b, arg...), nil
+}
+
+func minikubeBinary() (string, error) {
+	execName := "minikube"
+	if found, _ := util.DetectWSL(); found {
+		execName = "minikube.exe"
+	}
+	filename, err := exec.LookPath(execName)
+	if err != nil {
+		return "", errors.New("unable to find minikube executable. Please add it to PATH environment variable")
+	}
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return "", fmt.Errorf("unable to find minikube executable. File not found %s", filename)
+	}
+	return filename, nil
+}
+
+func matchNodeLabel(profileName string) (bool, error) {
+	client, err := k8s.Client()
+	if err != nil {
+		return false, fmt.Errorf("getting Kubernetes client: %w", err)
+	}
+	opts := v1.ListOptions{
+		LabelSelector: fmt.Sprintf("minikube.k8s.io/name=%s", profileName),
+		Limit:         100,
+	}
+	l, err := client.CoreV1().Nodes().List(opts)
+	if err != nil {
+		return false, fmt.Errorf("listing nodes with matching label: %w", err)
+	}
+	return l != nil && len(l.Items) > 0, nil
+}
+
+// matchProfileAndServerURL checks if kubecontext matches any valid minikube profile
+// and for selected drivers if the k8s server url is same as any of the minikube nodes IPs
+func matchProfileAndServerURL(kubeContext string) (bool, error) {
+	config, err := k8sConfigFunc()
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes config: %w", err)
+	}
+	apiServerURL, _, err := rest.DefaultServerURL(config.Host, config.APIPath, schema.GroupVersion{}, false)
+
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes server url: %w", err)
+	}
+
+	logrus.Debugf("kubernetes server url: %s", apiServerURL)
+
+	ok, err := matchServerURLFor(kubeContext, apiServerURL)
+	if err != nil {
+		return false, fmt.Errorf("checking minikube node url: %w", err)
+	}
+	return ok, nil
+}
+
+func matchServerURLFor(profileName string, serverURL *url.URL) (bool, error) {
```

Comment:
Same here, log error and return only bool.

Reply:
I'll end up having a bunch of log messages in these functions. I think I'll leave it as is and log only in the top method